python_modules/dagster/dagster_tests/cli_tests/test_api_commands.py

```diff
@@ -59,7 +60,67 @@ def test_execute_run():
                 [input_json],
             )
 
-        assert "PIPELINE_SUCCESS" in result.stdout, "no match, result: {}".format(result)
+            assert "PIPELINE_SUCCESS" in result.stdout, "no match, result: {}".format(result.stdout)
+
+            # Framework errors (e.g. running a run that has already run) still result in a non-zero error code
+            result = runner.invoke(api.execute_run_command, [input_json])
+            assert result.exit_code == 0
+
+
+def test_execute_run_fail_pipeline():
+    with get_bar_repo_handle() as repo_handle:
+        pipeline_handle = PipelineHandle("fail", repo_handle)
+        runner = CliRunner()
+
+        with instance_for_test(
+            overrides={
+                "compute_logs": {
+                    "module": "dagster.core.storage.noop_compute_log_manager",
+                    "class": "NoOpComputeLogManager",
+                }
+            }
+        ) as instance:
+            instance = DagsterInstance.get()
+            run = create_run_for_test(instance, pipeline_name="foo", run_id="new_run")
+
+            input_json = serialize_dagster_namedtuple(
+                ExecuteRunArgs(
+                    pipeline_origin=pipeline_handle.get_python_origin(),
+                    pipeline_run_id=run.run_id,
+                    instance_ref=instance.get_ref(),
+                )
+            )
+
+            result = runner_execute_run(
+                runner,
+                [input_json],
+            )
+            assert result.exit_code == 0
+
+            assert "RUN_FAILURE" in result.stdout, "no match, result: {}".format(result)
+
+            run = create_run_for_test(
+                instance, pipeline_name="foo", run_id="new_run_raise_on_error"
+            )
+
+            input_json_raise_on_failure = serialize_dagster_namedtuple(
+                ExecuteRunArgs(
+                    pipeline_origin=pipeline_handle.get_python_origin(),
+                    pipeline_run_id=run.run_id,
+                    instance_ref=instance.get_ref(),
+                    set_exit_code_on_failure=True,
+                )
+            )
+
+            result = runner.invoke(api.execute_run_command, [input_json_raise_on_failure])
+
+            assert result.exit_code != 0, str(result.stdout)
+
+            assert "RUN_FAILURE" in result.stdout, "no match, result: {}".format(result)
+
+            # Framework errors (e.g. running a run that has already run) also result in a non-zero error code
+            result = runner.invoke(api.execute_run_command, [input_json_raise_on_failure])
+            assert result.exit_code != 0, str(result.stdout)
```

Comment:
@alangenfeld this line is an example of a test case that would fail if we relied on "a failure event was yielded during this output" as the sole criteria for returning a non-zero error code